### DISCUSSION TITLE
Provisions for com.google.storage.v2

### DISCUSSION THREAD

**Discussion by smahad786** [fsouza/fake-gcs-server#1275] [Q&A]
The fake-gcs-server works for HTTP 1.x, I would like to see whether it is possible to get this working for com.google.storage.v2. When I try to use com.google.storage.v2, I get http2 exception. Any ideas how or directions on this will be a great help

**Comment by fsouza** [comment]
What's `com.google.storage.v2`? I assume it's the Java SDK given the naming convention, but can you confirm? Also, can you provide details on the exception?

**Comment by smahad786** [comment]
Link for com.google.storage.v2 - https://cloud.google.com/java/docs/reference/google-cloud-storage/latest/com.google.storage.v2

When I spin the fakeServer container in 9023 port and use com.google.storage.v2.StorageGrpc to invoke any method like listBuckets(), it just gives me a http2 exception in the docker logs, just a one liner. However, when I use the com.google.cloud.storage.Storage and call listBuckets(), I am able to run. 


I launched the docker container via command line at port 9023 and created the following:

    channel = ManagedChannelBuilder
        .forAddress("localhost", 9023)
        .usePlaintext().build();

    asyncStub = StorageGrpc
        .newStub(channel);

and then using the stub to run the listBuckets() for my Unit Test. 
Hope this helps

**Comment by smahad786** [comment]
The actual error message in OnError() in responseObserver and the logs from the docker logs listed below. I realized that  client was making a HTTP2 call and the sever can only accommodate HTTP 1.x call. 

error listing Bucket io.grpc.StatusRuntimeException: INTERNAL: http2 exception

% docker run  --name fake-gcs-server -p 4443:4443 -v $HOME/data:/data dockerproxy.aexp.com/fsouza/fake-gcs-server -scheme http
time="2023-08-02T22:05:28Z" level=info msg="server started at http://0.0.0.0:4443"
time="2023-08-02T22:06:22Z" level=info msg="172.17.0.1 - - [02/Aug/2023:22:06:22 +0000] \"PRI * HTTP/2.0\" 404 19"
time="2023-08-02T22:06:22Z" level=info msg="172.17.0.1 - - [02/Aug/2023:22:06:22 +0000] \"PRI * HTTP/2.0\" 404 19

**Reply by fsouza** [reply]
Although I haven't tested it, Go _should_ bind to HTTP/2 automatically if you use HTTPS, and as of #1138, fake-gcs-server implements the gRPC API. Can you give that a shot?

If you'd like to use gRPC with an insecure channel, can you open an issue for that, and also confirm that the Java SDK supports it?
